### PR TITLE
Added logsumexp decomposition

### PR DESCRIPTION
**Pull request by Chillee**
Pretty simple.

cc: @jansel who mentioned this.

### CODE REVIEW

**Review by jansel — APPROVED** [review]
Thanks!

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -1200,3 +1203,23 @@ def stack(tensors: List[Tensor], dim: int = 0) -> Tensor:
         return out.view(result_sizes)
     else:
         return torch.cat(get_stack_inputs(tensors, wrapped_dim), dim)
+
+
+def _squeeze_multiple(self, dims):
+    ndims = self.dim()
+    for idx in range(ndims - 1, -1, -1):
```
Comment (jansel) [inline torch/_decomp/decompositions.py:1210]: is this correct from dim=-2?

File: torch/_decomp/decompositions.py [hunk 2]
```diff
@@ -1200,3 +1203,23 @@ def stack(tensors: List[Tensor], dim: int = 0) -> Tensor:
...
+        if idx in dims:
+            self = self.squeeze(idx)
+    return self
+
+
+@register_decomposition(aten.logsumexp.default)
+@pw_cast_for_int_to_real
+def logsumexp(self: Tensor, dim: List[int], keepdim: bool = False) -> Tensor:
+    if self.numel() == 0:
```
Comment (jansel) [inline torch/_decomp/decompositions.py:1219]: This will add a guard that has to compute a product, do we want that?

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_decomp/decompositions.py (hunk 1 above)
Comment (Chillee) [reply torch/_decomp/decompositions.py:1210]: I need to add a wrap dim here.

Comment (Chillee) [reply torch/_decomp/decompositions.py:1210]: A bit surprised it's not tested in OpInfos - added a test there.

File: torch/_decomp/decompositions.py (hunk 2 above)
Comment (Chillee) [reply torch/_decomp/decompositions.py:1219]: Filed an issue about it: https://github.com/pytorch/pytorch/issues/77232

**Review by jansel — APPROVED** [review]

File: torch/_decomp/decompositions.py [hunk 3]
```diff
@@ -656,10 +656,13 @@ def logit_backward(
 
 @register_decomposition(aten.native_dropout)
 @pw_cast_for_opmath
-def native_dropout_decomposition(input: Tensor, p: float, train: Optional[bool]):
-    bool_mask = torch.rand_like(input) < p
-    res = bool_mask * input * float(1.0 / p)
-    return [res, bool_mask]
+def native_dropout(input: Tensor, p: float, train: Optional[bool]):
```
Comment (vadimkantorov) [inline torch/_decomp/decompositions.py:659]: should it use `training` instead of `train` for consistency with higher-level regular https://pytorch.org/docs/stable/generated/torch.nn.functional.dropout.html?highlight=dropout#torch.nn.functional.dropout ?

Comment (Chillee) [reply torch/_decomp/decompositions.py:659]: In general I’ve been going for consistency with the aten schema. I could buy that for ops we’re unifying with the torch api we should use the torch names though (which doesn’t apply here).

File: torch/_decomp/decompositions.py [hunk 4]
```diff
@@ -656,10 +656,13 @@ def logit_backward(
...
+    if train:
+        bool_mask = torch.rand_like(input) < p
+        res = bool_mask * input * float(1.0 / p)
+        return [res, bool_mask]
+    else:
+        return [input, torch.ones_like(input)]
```
[the remark this answers is not in the document]
Comment (Chillee) [reply torch/_decomp/decompositions.py]: cool, done - hopefully it passes CI?
